ambry-store/src/main/java/com.github.ambry.store/CompactionDetails.java

```diff
@@ -0,0 +1,153 @@
+   *                      were created.
+   * @param logSegmentsUnderCompaction the names of the {@link LogSegment} under compaction.
+   * @param extraSegmentName the name of the segment which will provide the "extra" data. Can be {@code null} or empty
+   *                         if no extra data is required.
+   * @param swapSpaceCount the number of swap spaces that will be required for compaction.
+   */
+  CompactionDetails(long referenceTime, List<String> logSegmentsUnderCompaction, String extraSegmentName,
+      int swapSpaceCount) {
+    this.referenceTime = referenceTime;
+    this.logSegmentsUnderCompaction = logSegmentsUnderCompaction;
+    this.extraSegmentName = extraSegmentName;
+    this.swapSpaceCount = swapSpaceCount;
+  }
+
+  /**
+   * @param stream the serialized version of the {@link CompactionDetails} that has to be loaded.
+   * @return a representation of all the details required for compaction from the given {@code stream}.
+   * @throws IllegalArgumentException if the version is not recognized
+   * @throws IOException if there is an I/O error reading from the stream
+   */
+  static CompactionDetails fromBytes(DataInputStream stream) throws IOException {
+    CompactionDetails details;
+    short version = stream.readShort();
+    switch (version) {
+      case VERSION_0:
+        long referenceTime = stream.readLong();
+        int segmentCount = stream.readInt();
+        List<String> logSegmentsUnderCompaction = new ArrayList<>();
+        for (int i = 0; i < segmentCount; i++) {
+          logSegmentsUnderCompaction.add(Utils.readIntString(stream));
+        }
+        String extraLogSegmentName = Utils.readIntString(stream);
+        if (extraLogSegmentName != null && extraLogSegmentName.isEmpty()) {
+          extraLogSegmentName = null;
+        }
+        int swapSpaceCount = stream.readInt();
+        details = new CompactionDetails(referenceTime, logSegmentsUnderCompaction, extraLogSegmentName, swapSpaceCount);
+        break;
+      default:
+        throw new IllegalArgumentException("Unrecognized version: " + version);
+    }
+    return details;
+  }
+
+  /**
+   * @return the epoch time to use to get the same results from the {@link BlobStore} as when the details were created
+   */
+  long getReferenceTime() {
+    return referenceTime;
+  }
+
+  /**
+   * @return the names of the segments under compaction.
+   */
+  List<String> getLogSegmentsUnderCompaction() {
+    return logSegmentsUnderCompaction;
+  }
+
+  /**
+   * @return the name of the segment which will provide the "extra" data. Can be {@code null} or empty if no extra data
+   * is required.
+   */
+  String getExtraSegmentName() {
+    return extraSegmentName;
+  }
+
+  /**
+   * @return the number of swap spaces that will be required for compaction.
+   */
+  int getSwapSpaceCount() {
+    return swapSpaceCount;
+  }
+
+  /**
```

Comment:
I don't want to add it in the javadoc because the serialized versions *may* contain private variables but included a description as a comment in the function. Is that ok?